web/packages/design/src/Alert/Alert.tsx

```diff
@@ -245,14 +250,23 @@ const OuterContainer = styled.div<AlertPropsWithRequiredKind>`
 `;
 
 /** Renders a transparent color overlay. */
-const InnerContainer = styled.div<AlertPropsWithRequiredKind>`
+const InnerContainer = styled.div<
+  Pick<WithRequired<AlertProps, 'kind' | 'alignItems'>, 'kind' | 'alignItems'>
+>`
   padding: 12px 16px;
   overflow: auto;
   word-break: break-word;
   display: flex;
-  align-items: center;
+  align-items: ${p => p.alignItems};
 
   ${backgroundColor}
+
+  // By default, <a> within an Alert has white color which has bad contrast on certain kinds of
+  // alerts, like Warning for example. This is an attempt to at least make them visible using the
+  // same color as the text.
+  & ${Link}, & a {
+    color: inherit;
```

Comment:
Ugh, I can't do this without risking breaking someone's UI if we ever used something like `<ButtonPrimary as="a">` within an alert.

Reply:
Fixed by specifically checking against `<Button>`.